modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/endpointComponents/PathParameterConfiguration.tsx

```diff
@@ -0,0 +1,187 @@
+	>();
+
+	const [pathParameterOptions, setPathParameterOptions] = useState<
+		SelectOption[]
+	>([]);
+
+	const handleSelectPathParameter = (value: string) => {
+		setData((previousValue) => ({
+			...previousValue,
+			pathParameter: value,
+		}));
+
+		setSelectedPathParameter(
+			pathParameterOptions.find((option) => option.value === value)
+		);
+	};
+
+	useEffect(() => {
+		setSelectedPathParameter(
+			pathParameterOptions.find(
+				(option) => option.value === data.pathParameter
+			)
+		);
+	}, [pathParameterOptions, data.pathParameter]);
+
+	const fetchSchemaByAPISchemaID = (
+		schemaId: string
+	): Promise<APISchemaItem> => {
+		return fetchJSON<APISchemaItem>({
+			input: `/o/headless-builder/schemas/${schemaId}`,
+		});
+	};
+
+	const fetchObjectDefinitionsByExternalReferenceCode = (
+		externalReferenceCode: string
+	): Promise<ObjectDefinition[]> => {
+		return getAllItems<ObjectDefinition>({
+			filter: `label eq 'ID' or label eq 'External Reference Code' or unique eq true`,
+			url: `/o/object-admin/v1.0/object-definitions/by-external-reference-code/${externalReferenceCode}/object-fields`,
+		});
+	};
+
+	const getObjectFieldLabel = (objectField: ObjectDefinition): string => {
+		if (
+			typeof objectField.label[
+				defaultLanguageId as keyof LocalizedValue<string>
+			] !== 'undefined'
+		) {
+			return objectField.label[
+				defaultLanguageId as keyof LocalizedValue<string>
+			]!;
+		}
+		else if (Object.keys(objectField.label).length !== 0) {
+			return Object.values(objectField.label)[0];
+		}
+		else {
+			return objectField.name;
+		}
+	};
+
+	const fetchPathParameterOptions = useCallback((schemaId: string) => {
+		fetchSchemaByAPISchemaID(schemaId).then((response) => {
+			fetchObjectDefinitionsByExternalReferenceCode(
+				response.mainObjectDefinitionERC
+			).then((response) => {
+				const options = response
+					? response.map((objectField) => ({
+							label: getObjectFieldLabel(objectField),
+							value: objectField.name,
+					  }))
+					: [];
+
+				if (options.length) {
+					setPathParameterOptions(options);
+				}
+			});
+		});
+	}, []);
+
+	useEffect(() => {
+		if (data.r_responseAPISchemaToAPIEndpoints_c_apiSchemaId) {
+			fetchPathParameterOptions(
+				data.r_responseAPISchemaToAPIEndpoints_c_apiSchemaId?.toString()
+			);
+		}
+	}, [
+		data.r_responseAPISchemaToAPIEndpoints_c_apiSchemaId,
+		fetchPathParameterOptions,
+	]);
+
+	return (
+		<>
+			{selectedResponseBodySchema ? (
+				<ClayForm.Group>
+					<label htmlFor="selectTrigger">
+						{Liferay.Language.get('path-parameter-property')}
+
+						<span className="ml-1 reference-mark text-warning">
+							<ClayIcon symbol="asterisk" />
+						</span>
+					</label>
+
+					<Select
+						disabled={false}
+						dropDownSearchAriaLabel={Liferay.Language.get(
+							'search-for-a-parameter-property-or-use-the-arrow-keys-to-navigate-and-select-a-parameter-property-from-the-list'
+						)}
+						onClick={handleSelectPathParameter}
```

Comment:
add an `id= "selectTrigger"` to this element, so the `htmlFor` for the label works.

Reply:
This is `id= "selectTrigger` is added automatically inside of the `Select` component :smile: